client/src/Components/Layouts/Captains/EditCaptain/editCaptain.jsx

```diff
@@ -0,0 +1,301 @@
+                  rules: [
+                    { required: true, message: "يرجى ملئ الحقل بحروف ", pattern: /^([أ-يa-z]|\s)+$/ }
+                  ]
+                })(<Input type="text" id="name"
+                />)}
+              </Form.Item>
+              <Form.Item label="البريد">
+                {getFieldDecorator("email", {
+                  rules: [
+                    {
+                      required: true,
+                      message: "يرجى ملئ الحقل بريد الكتروني",
+                      pattern: /.+\@.+\..+/
+                    }
+                  ]
+                })(<Input type="email" id="email" />)}
+              </Form.Item>
+              <Form.Item label="كلمة المرورالجديدة">
+                {getFieldDecorator("password", {
+                  rules: [
+                    {
+                    }
+                  ]
+                })(<Input type="password" id="password" />)}
+              </Form.Item>
+              <Form.Item label="الهاتف">
+                {getFieldDecorator("phone", {
+                  rules: [
+                    {
+                      required: true,
+                      message: "الرجاء ملئ الحقل بارقام",
+                      pattern: /^\+?[0-9]{10,12}$/
+                    }
+                  ]
+                })(<Input type="text" id="phone" />)}
+              </Form.Item>
+
+              <Form.Item label="العنوان" dir="ltr">
+                {getFieldDecorator("address", {
+                  rules: [
+                    { required: true, message: "يرجى ملئ الحقل بحروف ", pattern: /^[أ-يa-z]*$/ }
+                  ]
+                })(<Input type="text" id="address" />)}
+              </Form.Item>
+            </div>
+          </Form>
+        </Modal>
+      );
+    }
+  }
+);
+
+class CollectionsPage extends React.Component {
+  state = {
+    visible: false,
+  };
+
+  showModal = () => {
+    this.setState({ visible: true });
+  };
+
+  handleCancel = () => {
+    this.setState({ visible: false });
+  };
+
+
+  handleCreate = () => {
+    const form = this.formRef.props.form;
+    form.validateFields((err, values) => {
+      if (err) {
+        return;
+      }
+      else {
+        const { IDNumber, address, email, licenceNumber, name, password, phone, status, file } = values;
+        const formData = new FormData();
+        if (!file.fileList) {
+          formData.append('file', file);
+        } else {
+          formData.append('file', file.fileList[0].originFileObj);
+        }
+        formData.append('IDNumber', IDNumber);
+        formData.append('address', address);
+        formData.append('email', email);
+        formData.append('licenceNumber', licenceNumber);
+        formData.append('name', name);
+        formData.append('password', password);
+        formData.append('phone', phone);
+        formData.append('status', status);
+        const id = 6;
+        // const { id } = this.props.id;
+        fetch(`/api/v1/putCaptain/${id}`, {
```

Comment:
@naremanhilles It's not something you need to change right now, but an idea for next time: It might be better code structure to have one component responsible for all of the network requests.
Right now, CollectionsPage does the PUT to the backend, but CollectionCreateForm does the GET to the backend.  Ideally, one of them would be doing both PUT and GET, and the other wouldn't have to worry about doing network requests.